build/metadata.py

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+# coding: utf-8
+"""
+Process manuscript metadata and author metadata
+"""
+import collections
+import pathlib
+
+import yaml
+
+
+def get_metadata_object(path):
+    """
+    Load all of the metadata from a yaml file, & return.
+
+    :param path: pathlib.Path to a metadata YAML file.
+    :return: structured metadata object.
+    """
+    class metadata(object):
```

Comment:
I am sure. The object being returned here is not meant to have key-value pairs added to it, so a dictionary type is overkill. While a dictionary type might provide similar functionality, it also means that membership is typically determined at run time, and since python doesn't have static or const, widely used entry-level static code analysis tools (e.g., pylint) won't catch the erroneous call in the following scenario:
```python
metadata['header_block']
metadata['header_blck'] # error at run time
```
Whereas in the following scenario they do:
```python
metadata.header_block
metadata.header_blck # easily catchable before run time
```
Since there is no reason to extend the contents of the metadata dict beyond what would be found in the file itself (I imagine that if you wanted to have empty optional fields you would want them represented by ```None```), and since not using a dict enhances ability to debug code, I decided to not use a dict.